src/Adapter/Order/CommandHandler/AddCartRuleToOrderHandler.php

```diff
@@ -63,7 +64,7 @@ public function handle(AddCartRuleToOrderCommand $command): void
         switch ($command->getCartRuleType()) {
             // Percent type
             case OrderDiscountType::DISCOUNT_PERCENT:
-                if ($discountValue < 100) {
+                if ($discountValue < 100 && $discountValue >= 0) {
```

Comment:
Thank you for this PR, I wasn't aware of this file while dealing with same issue :) .
but, I am a bit confused about `0` and `100` values.
From `order page` legacy: 
* possible to create percent discount with value 0%
* impossible to create percent discount with value 100%
From `catalog add cart rule` legacy:
* imPossible to create percent discount with value 0%.
* possible to create percent discount with value  100%
If you create discount from  `order page` legacy with 99% value, it is possible to edit it to 100% in  `catalog edit cart rule`. So what are the specs for this?

Reply:
Thanks @arouiadib for the discovery. The logic should respect the catalog and cart rule behaviors such as :
- impossible to create a percent discount with value of 0%.
- possible to create a percent discount with value of 100%

@atomiix Could you fixed in this PR ?

Reply:
Indeed this need to be confirmed and specified with the product team. Besides would it be possible to change the detect and detect the exception cases rather than managing this in an `else` statement.
Something like:
```php
if ($discountValue > 100 || $discountValue <= 0) {
    throw new NegativeCartRuleValueException(
        'Reduction percentage must be between 0% and 100%.',
        NegativeCartRuleValueException::NEGATIVE_PERCENTAGE
    );
}
// Do the action here
```

This would reduce the indentation which makes the code more readable And the condition for error is more straight forward.

Oh and I can't comment the appropriate line, but I think `$discountValue = (float) str_replace(',', '.', $command->getDiscountValue());` is useless since the command contains a `Number` we need to get rid of these nasty/useless `str_replace`

Reply:
@arouiadib @jolelievre @MatShir I'm taking everything into account, I'm just waiting for new exception messages before pushing new changes ;)